### PR TITLE
Add analyze command

### PR DESCRIPTION
**Pull request by khernyo**
| Q               | A
| --------------- | ---
| Bug fix?        | no
| New feature?    | yes
| API breaks?     | no
| Deprecations?   | no
| Related tickets |
| License         | Apache 2.0

### What's in this PR?
image

### To Do
- [x] go.mod update
- [x] multi cluster support
- [x] depends on banzaicloud/backyards#388

### Postponed
- support for validations through graphql api

### CODE REVIEW

File: internal/cli/cmd/analyze/cmd.go [hunk 1]
```diff
@@ -0,0 +1,201 @@
...
+	if err != nil {
+		panic(err)
+	}
+
+	configs := map[string]pods.SidecarCheckConfig{
+		api.CPClusterID: masterConfig,
+	}
+
+	mTLSUnsetCheckConfig := destinationrules.MTLSUnsetCheckConfig{
+		AutoMTLSEnabled: true,
```
Comment (Laci21) [inline internal/cli/cmd/analyze/cmd.go:90]: Shouldn't this be fetched from the api server?

File: internal/cli/cmd/analyze/cmd.go [hunk 2]
```diff
@@ -0,0 +1,201 @@
...
+						printKeyValueInfo(fmt.Sprintf("\t\t\t%s: ", k), fmt.Sprint(v))
+					}
+				}
+			}
+		}
+	}
+
+	switch errorCount {
+	case 0:
+		cli.Logger().Info("0 validation errors found")
```
Comment (Laci21) [inline internal/cli/cmd/analyze/cmd.go:145]:
```suggestion
		cli.Logger().Info("0 validation error found")
```

File: pkg/cli/cmd/root.go [hunk 3]
```diff
@@ -25,6 +25,7 @@ import (
 	"emperror.dev/errors"
 	"github.com/AlecAivazis/survey/v2"
 	"github.com/MakeNowJust/heredoc"
+	"github.com/banzaicloud/backyards-cli/internal/cli/cmd/analyze"
```
Comment (Laci21) [inline pkg/cli/cmd/root.go]: imho internal import should be lower

File: internal/cli/cmd/analyze/cmd.go (hunk 2 above)
Comment (khernyo) [reply internal/cli/cmd/analyze/cmd.go:145]: "0 validation errors" is the grammatically correct form

File: pkg/cli/cmd/root.go (hunk 3 above)
Comment (khernyo) [reply pkg/cli/cmd/root.go]: fixed

File: internal/cli/cmd/analyze/cmd.go (hunk 1 above)
Comment (khernyo) [reply internal/cli/cmd/analyze/cmd.go:90]: I guess so. This is copy-pasta code, I will fix it later

File: internal/cli/cmd/analyze/cmd.go (hunk 2 above)
Comment (Laci21) [reply internal/cli/cmd/analyze/cmd.go:144]: oh, good to know, thanks! :)

**Pull request merged**